### PR TITLE
Add polkadex fungible assets

### PR DESCRIPTION
**Pull request by ksr30**

### CODE REVIEW

File: pallets/Polkadex_Fungible_Assets/src/lib.rs [hunk 1]
```diff
@@ -0,0 +1,320 @@
...
+                        Ok(())
+                    })
+                })
+            }
+            Some(_) => Err(<Error<T>>::NoPermissionToBurn),
+            None => Err(<Error<T>>::AssetIdNotExists),
+        }
+    }
+
+    pub fn transfer_native(
```
Comment (Gauthamastro) [inline pallets/Polkadex_Fungible_Assets/src/lib.rs:300]: Why do we need this function?

File: pallets/Polkadex_Fungible_Assets/src/lib.rs [hunk 2]
```diff
@@ -0,0 +1,320 @@
...
+
+        /// Modify Token Registration
+        #[weight = 10000]
+        pub fn modify_token_deposit_amount(origin, pdx_amount: T::Balance) -> DispatchResult {
+            T::GovernanceOrigin::ensure_origin(origin)?;
+            <FixedPDXAmount<T>>::put::<T::Balance>(pdx_amount);
+            Self::deposit_event(RawEvent::TokenDepositModified(pdx_amount));
+            Ok(())
+        }
+
```
Comment (Gauthamastro) [inline pallets/Polkadex_Fungible_Assets/src/lib.rs:242]: We need one more dispatchable functions for claiming the vested tokens

fn claim( vestingID): something like this, We need to identify each vesting using some identifiers

File: pallets/Polkadex_Fungible_Assets/src/lib.rs [hunk 3]
```diff
@@ -0,0 +1,320 @@
...
+                        max_supply: T::Balance,
+                        mint_account: Option<T::AccountId>,
+                        burn_account: Option<T::AccountId>,
+                        existenial_deposit: T::Balance) -> DispatchResult {
+            let who: T::AccountId = ensure_signed(origin)?;
+            ensure!(!orml_tokens::TotalIssuance::<T>::contains_key(asset_id), Error::<T>::AssetIdAlreadyExists);
+            ensure!(!<InfoAsset<T>>::contains_key(&asset_id), Error::<T>::AssetIdAlreadyExists);
+            let tresury_account = T::TreasuryAccountId::get();
+            let amout_to_trasfer: T::Balance = FixedPDXAmount::<T>::get();
+//		    orml_currencies::NativeCurrencyOf::<T>::transfer(&who, &tresury_account, amout_to_trasfer);
```
Comment (Gauthamastro) [inline pallets/Polkadex_Fungible_Assets/src/lib.rs:165]: orml_currencies's transfer_native function here

File: pallets/Polkadex_Fungible_Assets/src/lib.rs [hunk 4]
```diff
@@ -0,0 +1,320 @@
+#![cfg_attr(not(feature = "std"), no_std)]
+
```
Comment (Gauthamastro) [inline pallets/Polkadex_Fungible_Assets/src/lib.rs:2]: Add GPL v3 license and a copyright notice here

File: pallets/Polkadex_Fungible_Assets/src/lib.rs [hunk 5]
```diff
@@ -0,0 +1,320 @@
...
+            let vesting_info = VestingInfo::from(amount, rate, current_block_no);
+            <InfoVesting<T>>::insert(account, vesting_info);
+            Ok(())
+        }
+
+        /// Set Metadata
+        #[weight = 10000]
+        pub fn set_metadata_fungible(origin, asset_id: T::CurrencyId, metadata: AssetMetadata) -> DispatchResult {
+            let who: T::AccountId = ensure_signed(origin)?;
+            ensure!(<InfoAsset<T>>::contains_key(&asset_id), <Error<T>>::AssetIdNotExists);
```
Comment (Gauthamastro) [inline pallets/Polkadex_Fungible_Assets/src/lib.rs:194]: Ensure the vectors of metadata are bounded.

File: primitives/src/lib.rs [hunk 6]
```diff
@@ -1,4 +1,4 @@
 #![cfg_attr(not(feature = "std"), no_std)]
 
-pub mod engine;
 pub mod assets;
+pub mod engine;
```
Comment (Gauthamastro) [inline primitives/src/lib.rs:4]: Remove engine, and delete engine file

File: runtime/src/lib.rs [hunk 7]
```diff
@@ -121,235 +127,259 @@ pub const DAYS: BlockNumber = HOURS * 24;
...
+    {
+        System: frame_system::{Module, Call, Config, Storage, Event<T>},
+        RandomnessCollectiveFlip: pallet_randomness_collective_flip::{Module, Call, Storage},
+        Timestamp: pallet_timestamp::{Module, Call, Storage, Inherent},
+        Aura: pallet_aura::{Module, Config<T>},
+        Grandpa: pallet_grandpa::{Module, Call, Storage, Config, Event},
+        Balances: pallet_balances::{Module, Call, Storage, Config<T>, Event<T>},
+        TransactionPayment: pallet_transaction_payment::{Module, Storage},
+        Sudo: pallet_sudo::{Module, Call, Config<T>, Storage, Event<T>},
+        PolkadexFungibleAssets: polkadex_fungible_assets::{Module, Call, Event<T>},
```
Comment (Gauthamastro) [inline runtime/src/lib.rs:378]: PolkadexFungibleAssets should be coming after Tokens and Currencies pallet

**Review by Gauthamastro — CHANGES_REQUESTED** [review]

File: pallets/Polkadex_Fungible_Assets/src/lib.rs (hunk 1 above)
Comment (ayushmishra2005) [reply pallets/Polkadex_Fungible_Assets/src/lib.rs:346]: @ksr30

**Review by Gauthamastro — APPROVED** [review]

**Pull request merged**